zenoh_cpp_vendor/CMakeLists.txt

```diff
@@ -35,4 +35,15 @@ ament_vendor(zenoh_c_vendor
 
 ament_export_dependencies(zenohc)
 
+ament_vendor(zenoh_cpp_vendor
```

Comment:
I think we need to compile `zenoh_cpp_vendor`.

Reply:
Right, my understanding is that the way this works is that `zenoh-cpp` needs either `zenoh-c` or `zenoh-pico` as a "backend" to compile against.  So we need to keep both `zenoh_c_vendor` and `zenoh_cpp_vendor`.

We could consider making this two separate vendor packages, and we might want to do that in the future to enable the easier use of `zenoh-pico` as the backend.  But I think this is OK as-is for now.

Reply:
> We could consider making this two separate vendor packages, and we might want to do that in the future to enable the easier use of zenoh-pico as the backend. But I think this is OK as-is for now.

One challenge with that would be keeping the commit hashes between the zenoh_c_vendor and zenoh_cpp_vendor compatible. 

Should we just rename this package to `zenoh_vendor` and export c and cpp targets?

Reply:
I think we should leave this as `zenoh_cpp_vendor` for now; after all, that is the API we are working against.  In the future, I think we may want to end up having 3 separate packages: `zenoh_c_vendor`, `zenoh_cpp_vendor`, and `zenoh_pico_vendor`.  But it isn't clear to me how the user would choose between the C and the pico backends, so that would need further thought.